tests/test_mig_unittest_testcore.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+#
+# --- BEGIN_HEADER ---
+#
+# addheader - add license header to all code modules.
+# Copyright (C) 2003-2024  The MiG Project by the Science HPC Center at UCPH
+#
+# This file is part of MiG.
+#
+# MiG is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# MiG is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+# USA.
+#
+# --- END_HEADER ---
+#
+
+"""Temporary wrapper allowing the inclusion of testcore into the automated suite."""
+
+import importlib
+import os
+import sys
+
+sys.path.append(os.path.realpath(os.path.join(os.path.dirname(__file__), ".")))
```

Comment:
What is this odd construct supposed to do? :)

Can we please converge on applying `realpath` or better `abspath` on `__file__` before N `dirname`s and avoid the "." and ".." if it is in effect just the absolute grandparent/parent/directory path we want. I'll push another change doing that along with a bit more style cleanup.

Reply:
I'm sorry this is frustrating as much as it is.

To reiterate:
- you are seeing a PR that has been rebased many many times at this point, thus some of what we've agreed since wasn't applied - I apologise this one made it through. I hope it is clear that I have taken all our agreements on board
- when I first attempted this work, I had to iterate on this file without a bunch of the more recent machinery, and that included a bunch of infrastructure that obviated this line entirely. I was previously able to break execution simply by changing folders, so I patched it to avoid that. as per above this should no longer be necessary

I've tried to toe the line of careful and iterative improvement while not introducing things that we later remove again, but sometimes this isn't possible given the pre-existing code and I'm also fallible. My hope was to do a follow-up PR that swept all of these lines away, and being iterative is necessarily going to mean that we land code that is known not to be perfect and follow-up. That's obviously difficult to capture when just reviewing an isolated code change, but if you've suggestions about how to capture and/or communicate that better between author and reviewer I will try my best to follow suit.

Reply:
Yeah, sorry if I seemed a bit hot-headed on that one :-/ no offense meant. I had somehow convinced myself we had already converged on it once, but alas, I can see we only just _discussed_ that. I edited my comment a bit when I realized my mistake. 

Yet, the original comment was really as much about the fact that it uses only a single dot `'.'` here (i.e. same dir), whereas it is `'..'` in the others (parent dir). If this _worked_ it tells me that the path mangling is no longer needed. I changed it to '..' in my follow-up for consistency, but would really like them to go completely :-)